index.js

```diff
@@ -490,14 +490,37 @@ exports.init = function (sbot, config) {
 
     function next(accept) {
       getConfirm(invite_id, accept, function (err, confirm) {
-        if(!confirm)
-          connectFirst(invite, function (err, rpc) {
-            if(err) return cb(err)
-            rpc.peerInvites.confirm(accept, function (err, confirm) {
-              //TODO: store confirms for us in the state.
-              cb(err, confirm)
-            })
+        if (err) return cb(err)
+        if(confirm) {
+          console.error('ssb-peer-invites: Invite already confirmed')
+          return // mix: send something else back in cb for client side?
```

Comment:
There should definitely be a callback here, this is actually a memory leak, because muxrpc will wait for the cb forever.

it should callback `confirm` might be a good idea to add something to confirm to indicate that it's already confirmed.

Reply:
ok, fixing this to cb

I don't know the data type of confirm, so I'm just sending back a new Error

Reply:
that's not really as good because now you have to check the error message of that error, to check if it's the already confirmed thing... and checking errors messages is weird because it's never clear if an error message is part of the api or not. this function calls back confirm _anyway_ so checking if the invite is already confirmed and calling back with it makes this operation [idempotent](https://www.c-sharpcorner.com/article/idempotent-operation-in-distributed-system/) which is a very useful property that makes database or distributed systems much easier to reason about

Reply:
> this function calls back confirm anyway

Could you clarify? I want to help get this code done but I'm pretty confused by it :/

Reply:
@Powersource I mean, if the call works normally, it calls back the `confirm` object (actually the message the pub posted to acknowledge the invite acceptance, iirc). But if you call and it's was already confirmed, don't error, just call back the already existing `confirm` object. the call meaning: give me the `confirm` object, creating it if necessary.

sometimes you get an error, where you create the confirm object, but then the connection could drop before you receive it. It will error, but actually it worked, but you don't know that. So you retry a little later, this time it ought to look like it worked. It doesn't matter that it actually worked the first time, just make it so that it eventually looks like it worked.

Reply:
@mixmix ^^ bump

Reply:
@dominictarr winging this a bit, is this what you mean? https://github.com/ssbc/ssb-peer-invites/pull/23

Reply:
@Powersource yes, that is it exactly